### PR TITLE
Issue a warning instead of throwing when s3 zerocopy part metadata directory already exists

### PR DESCRIPTION
**Pull request by myrrc**
### Changelog category (leave one):
- Improvement

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Revert old behavior on zerocopy s3 metadata creation: if a directory exists (and checksums match), re-fetch it instead of throwing.

In 23.7 https://github.com/ClickHouse/ClickHouse/pull/50489 (backported to 23.3.10) an exception is thrown if part metadata already exists. One of our clients encountered this issue, so a fix is proposed: back to 23.3.4 solution where a warning is issued in a log. Throwing an exception makes zerocopy replication unusable.

### CODE REVIEW

File: src/Storages/MergeTree/MergeTreePartsMover.cpp [hunk 1]
```diff
@@ -227,12 +228,31 @@ MergeTreePartsMover::TemporaryClonedPart MergeTreePartsMover::clonePart(const Me
...
+            if (!existing_checksums.read(buf))
+                throw Exception(ErrorCodes::LOGICAL_ERROR,
+                    "Cannot clone part {} from '{}' to '{}': path '{}' already exists"
+                    "and checksums are corrupted",
+                    part->name, part->getDataPartStorage().getDiskName(), disk->getName(),
+                    fullPath(disk, path_to_clone + relative_path));
+
+            part->checksums.checkEqual(existing_checksums, /*don't uncompress data*/false);
+            // Easier to re-fetch than to re-init part storage from existing files
+            disk->removeRecursive(fs::path(path_to_clone) / relative_path / "");
```
Comment (myrrc) [inline src/Storages/MergeTree/MergeTreePartsMover.cpp:255]: Better solution would be to directly initialize new part storage from existing files, but I haven't found a suitable way to do it so far

**Comment by robot-clickhouse** [comment]
This is an automated comment for commit 4f36be3633b20f67036ef87dfb6eb91d9280b7f9 with description of existing statuses. It's updated for the latest CI running
The full report is available [here](https://s3.amazonaws.com/clickhouse-test-reports/53794/4f36be3633b20f67036ef87dfb6eb91d9280b7f9/ci_running.html)
The overall status of the commit is 🟡 pending

<table><thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody><tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td>🟡 pending</td></tr>
<tr><td>Mergeable Check</td><td>Checks if all other necessary checks are successful</td><td>🟢 success</td></tr>
</table>

**Comment by alexey-milovidov** [comment]
Zero-copy replication is experimental, so we don't need backporting.

**Comment by vdimir** [comment]
> Zero-copy replication is experimental, so we don't need backporting.

The problem is that we backported another patch that modified this line, and behavior in 23.3 was changed. https://github.com/ClickHouse/ClickHouse/pull/50489
Another thought is that the condition probably shouldn't happen again after that patch. Option `allow_remove_stale_moving_parts` should be enabled, though. And we still don't now for sure if it was the only bug that triggered this condition.

**Pull request closed without merging**